### PR TITLE
Bugfix: Skipping Service Packs (bsc#1158534)

### PR DESCRIPTION
**Pull request by taroth21**
### Description

Fix for https://bugzilla.suse.com/show_bug.cgi?id=1158534:  [doc] 5.2.1 Supported Upgrade Paths for SLE HA and SLE HA Geo - skip a service pack - supported? 

### Checklist
* Check all items that apply.

*Are backports required?*

- [ ] To maintenance/SLEHA12SP4
- [x] To maintenance/SLEHA15
- [x] To maintenance/SLEHA15SP1

### CODE REVIEW

File: xml/ha_migration.xml [hunk 1]
```diff
@@ -133,17 +133,28 @@
    migrate to.
   </para>
 
+  <para>
+   The &hasi; has the same supported upgrade paths as the underlying base system. For a complete
+   overview, see the section <link
+   xlink:href=" https://documentation.suse.com/sles/15-SP2/html/SLES-all/cha-upgrade-paths.html#sec-upgrade-paths-supported">
+    <citetitle>Supported Upgrade Paths to &sls; &productnumber;</citetitle></link> in the
```
[the remark this answers is not in the document]
Comment (lproven) [reply xml/ha_migration.xml:140]: Agreed. What do you think about this suggestion:

```
   overview, see the section <link xlink:href=
   "https://documentation.suse.com/sles/15-SP2/html/SLES-all/cha-upgrade-paths.html#sec-upgrade-paths-supported"
   ><citetitle>Supported Upgrade Paths to &sls; &productnumber;</citetitle></link> in the
```

...? (Trying to keep the quoted URL all on one line.)

File: xml/ha_migration.xml [hunk 2]
```diff
@@ -128,22 +128,33 @@
...
+   In addition, the following rules apply, as the &ha; cluster stack offers two methods for
+   upgrading the cluster:</para>
   <itemizedlist>
    <listitem>
     <para>
-     Cluster rolling upgrades are only supported within the same major release (from the
-     GA of a product version to the next service pack, and from one service pack
+     <emphasis>Cluster rolling upgrades</emphasis> are only supported within the same major
+     release (from the GA of a product version to the next service pack, and from one service pack
      to the next).
```
Comment (lproven) [inline xml/ha_migration.xml:152]:
```suggestion
     release (from one service pack to the next, or from the GA version of a product to SP1).
```

File: xml/ha_migration.xml [hunk 3]
```diff
@@ -554,6 +565,20 @@
...
     </tgroup>
    </informaltable>
+
+   <!--taroth 2020-04-27: the following note is copied from
+    https://github.com/SUSE/doc-sle/edit/maintenance/SLE15SP1/xml/sle_update_upgrading.xml,
+    ll. 57-70 and adjusted to the HA product-->
+   <note>
+    <title>Skipping Service Packs</title>
+    <para>
+     Consecutively installing all Service Packs is the easiest upgrade path.
```
Comment (lproven) [inline xml/ha_migration.xml:575]:
```suggestion
     The easiest upgrade path is consecutively installing all service packs.
```

Comment (lproven) [reply xml/ha_migration.xml:575]: We do not seem to be consistently capitalizing "Service Pack(s)" everywhere, so I suggest dropping it.

As a minimum, perhaps, capitalizing only the first usage in a particular book.

File: xml/ha_migration.xml [hunk 4]
```diff
@@ -554,6 +565,20 @@
...
+     For the &sle; 15 product line (GA and the following Service Packs) it is
```
Comment (lproven) [inline xml/ha_migration.xml:576]:
```suggestion
     For the &sle; 15 product line (GA and the following service packs) it is
```

File: xml/ha_migration.xml [hunk 5]
```diff
@@ -554,6 +565,20 @@
...
+     also supported to skip one service pack when upgrading. For example, it is
+     supported to upgrade from SLE&nbsp;HA&nbsp;15&nbsp;GA to 15&nbsp;SP2 or from
+     SLE&nbsp;HA&nbsp;15&nbsp;SP1 to 15&nbsp;SP3.
```
Comment (lproven) [inline xml/ha_migration.xml:579]:
```suggestion
     also supported to skip one service pack when upgrading. For example, upgrading from
     SLE&nbsp;HA&nbsp;15&nbsp;GA to 15&nbsp;SP2 or from SLE&nbsp;HA&nbsp;15&nbsp;SP1
     to 15&nbsp;SP3 is supported.
```

File: xml/ha_migration.xml [hunk 6]
```diff
@@ -133,17 +133,28 @@
...
+  <para>
+   In addition, the following rules apply, as the &ha; cluster stack offers two methods for
+   upgrading the cluster:</para>
   <itemizedlist>
    <listitem>
     <para>
-     Cluster rolling upgrades are only supported within the same major release (from the
-     GA of a product version to the next service pack, and from one service pack
+     <emphasis>Cluster rolling upgrades</emphasis> are only supported within the same major
+     release (from the GA of a product version to the next service pack, and from one service pack
```
[the remark this answers is not in the document]
Comment (taroth21) [reply xml/ha_migration.xml]: I'm not a big fan of formalparas but a good idea nevertheless. @tomschr However, I just noticed that the title of a formalpara does not allow any xrefs inside. Any idea why?

[the remark this answers is not in the document]
Comment (tomschr) [reply xml/ha_migration.xml]: > However, I just noticed that the title of a formalpara does not allow any xrefs inside. Any idea why?

@taroth21: Are you sure? I've tested it with Geekodoc v1.0.2.1 and you can add `<xref/>`s inside a `<title>`. Even in `formalpara/title`. I see no difference.

[the remark this answers is not in the document]
Comment (taroth21) [reply xml/ha_migration.xml]: Yes, you are right - on second try, it worked (not sure what the problem was). Fixed now.

File: xml/ha_migration.xml [hunk 7]
```diff
@@ -554,6 +564,20 @@
...
    </informaltable>
+
+   <!--taroth 2020-04-27: the following note is copied from
+    https://github.com/SUSE/doc-sle/edit/maintenance/SLE15SP1/xml/sle_update_upgrading.xml,
+    ll. 57-70 and adjusted to the HA product-->
+   <note>
+    <title>Skipping Service Packs</title>
+    <para>
+     The easiest upgrade path is consecutively installing all service packs.
+     For the &sle; 15 product line (GA and the following service packs) it is
```
Comment (lproven) [inline xml/ha_migration.xml:578]:
```suggestion
     For the &sle; 15 product line (GA and the subsequent service packs) it is
```

**Comment by taroth21** [comment]
Backported to the following versions:

- maintenance/SLEHA15SP1 (https://github.com/SUSE/doc-sleha/commit/b5596d7807)
- maintenance/SLEHA15 (https://github.com/SUSE/doc-sleha/commit/c57ee8aa0)

**Pull request merged**